src/main/java/com/direwolf20/buildinggadgets/client/gui/base/EntryList.java

```diff
@@ -0,0 +1,197 @@
+package com.direwolf20.buildinggadgets.client.gui.base;
+
+import com.mojang.blaze3d.platform.GlStateManager;
+import com.mojang.blaze3d.platform.GlStateManager.DestFactor;
+import com.mojang.blaze3d.platform.GlStateManager.SourceFactor;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.AbstractGui;
+import net.minecraft.client.gui.widget.list.ExtendedList;
+import net.minecraft.client.gui.widget.list.ExtendedList.AbstractListEntry;
+import net.minecraft.client.renderer.BufferBuilder;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.util.math.MathHelper;
+import org.lwjgl.opengl.GL11;
+
+import static org.lwjgl.opengl.GL11.*;
+
+/**
+ * A preset of {@link ExtendedList} where
+ * <ul>
+ * <li>No overlays will be drawn
+ * <li>No dirt background, but a gradient rectangle instead
+ * <li>Use {@link GL11#glScissor(int, int, int, int)} instead of overlay pieces to cover overflows
+ * <li>Allow selection
+ * <li>Ability to disable the transitioning black rectangles (override {@link #renderTransition(Tessellator,
+ * BufferBuilder, int, int)}
+ * </ul>
+ */
+public class EntryList<E extends AbstractListEntry<E>> extends ExtendedList<E> {
+
+    public static final int SCROLL_BAR_WIDTH = 6;
+
+    public EntryList(int left, int top, int width, int height, int slotHeight) {
+        super(Minecraft.getInstance(), width, height, top, top + height, slotHeight);
+        // Set left x and right x, somehow MCP gave it a weird name
+        this.setLeftPos(left);
+    }
+
+    @Override
+    public void render(int mouseX, int mouseY, float partialTicks) {
+        glEnable(GL_SCISSOR_TEST);
+        double guiScaleFactor = Minecraft.getInstance().mainWindow.getGuiScaleFactor();
+        glScissor((int) (getLeft() * guiScaleFactor),
+                (int) (Minecraft.getInstance().mainWindow.getHeight() - (getBottom() * guiScaleFactor)),
+                (int) (width * guiScaleFactor),
+                (int) (height * guiScaleFactor));
+        renderParts(mouseX, mouseY, partialTicks);
```

Comment:
You could have simply used `super.render()` here as the rest of the render code is basically the same as what you have added

Reply:
https://gist.github.com/MichaelHillcox/6dd7c28269599966ab4a5eabf5198273 see here for example of what I mean.

Reply:
![2019-09-02_12 04 35](https://user-images.githubusercontent.com/36975818/64130932-404cf400-cd7a-11e9-9769-cf225d4130c8.png)
![2019-09-02_12 04 45](https://user-images.githubusercontent.com/36975818/64130933-404cf400-cd7a-11e9-9948-9a7c33251591.png)
Well if we can live with the first one, this can be removed. But otherwise no.